### DISCUSSION TITLE
Diffs between protocol id and filenames of protocols

### DISCUSSION THREAD

**Discussion by Lauler** [swerik-project/the-swedish-parliament-corpus#20] [Q&A] [answered]
When working with the parliamentary corpus I noticed there are differences between how the protocol ids in the "preface" div element look like and how the xml filenames of the protocols are named. 

Examples:

* `prot-1966-höst-ak--30` -> `prot-1966--ak--030.xml`
* `prot-199899--4` -> `prot-199899--004.xml`

I realize left-padding with 0s in filenames ensures they are sorted as expected on filesystems, and certain characters might not be suitable in filenames. However, this represented a minor annoyance when working with the material! 

Any chance to have these harmonized? Or alternatively include a field in the xml with the filesystem-normalized id?

**Comment by ninpnin** [comment, marked as answer]
The XML ID in the TEI element corresponds to the new name.

**Reply by Lauler** [reply]
Wow, thanks!

I can't believe I didn't see this: https://github.com/swerik-project/riksdagen-records/blob/c60a8bccfab45c041b1a2576fa819120c8167be7/data/1966/prot-1966--ak--030.xml#L2C1-L2C70  

Good thing there's a Github discussions page to ask dumb questions :smile:

**Reply by MansMeg** [reply]
Alright! So everyone is happy now?

**Reply by Lauler** [reply]
Yes, everything fine. I grabbed the id from the wrong place.  

https://github.com/swerik-project/riksdagen-records/blob/c60a8bccfab45c041b1a2576fa819120c8167be7/data/1966/prot-1966--ak--030.xml#L31-L32

The KB id of the document is apparently placed in the preface. It was conveniently placed beside the docdate, so I grabbed it and used it thinking it was the same as your internal id. Might be a good idea to do what Väinö suggested below (move the KB id away from the preface).

**Comment by MansMeg** [comment]
We dont want to annoy the users. :) 

I would open an issue on this. The changes of filenames to include 0s what a previous issue fixed for @ljo . So the question is how to keep the filenames and reduce your annoyance. :)

**Comment by ninpnin** [comment]
Eventually we want to remove the preface and have a proper KB ID element, eg. with the idno

```xml
<idno type="kb-lab">prot-1966-höst-ak--30</idno>
```

See https://clarin-eric.github.io/parla-clarin/#TEI.idno
